arch/nios2/soc/nios2-qemu/soc.h

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#ifndef __SOC_H_
+#define __SOC_H_
+
+#include <system.h>
+
+#define UART_NS16550_PORT_0_BASE_ADDR		A_16550_UART_0_BASE
+#define UART_NS16550_PORT_0_IRQ			A_16550_UART_0_IRQ
+#define UART_NS16550_PORT_0_CLK_FREQ		A_16550_UART_0_FREQ
+
```

Comment:
is it possible to put these defines in board.h instead?

Reply:
This change is in line with folder or file structure with arch/nios2/soc/nios2f-zephyr/
I just want to make sure both the folders are aligned.

Reply:
Unless it's an off-chip UART integrated by the board, MMIO and IRQ stuff is part of the SoC design and thus should be in soc.h, yeah.

(Though of course this is an FPGA and in principle this can surely be changed at synthesis time and we should probably have yet another layer of abstraction if we really want to be pedantic.  We surely don't).

Reply:
Wasn't clear to me if the UART was concerned part of the SoC or not, if its part of the SoC than I'm fine with this.

Reply:
This needs to be at the SOC level, it's how the soft-cpu is wired up.